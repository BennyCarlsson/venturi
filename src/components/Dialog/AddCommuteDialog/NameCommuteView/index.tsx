import Paragraph from 'components/Paragraph'
import React from 'react'
import {
  CTAButtonTitleWrapper,
  Input,
  StyledButton,
  StyledClosedIconButton,
  StyledParagraph,
  StyledTitleHeading,
  TopCTAButtonTitleSubtitle,
} from './styles'

type NameCommuteViewProps = {
  createTrip: () => void
  goToChooseDestionationView: () => void
}

const NameCommuteView = ({
  createTrip,
  goToChooseDestionationView,
  ...props
}: NameCommuteViewProps) => {
  return (
    <React.Fragment {...props}>
      <TopCTAButtonTitleSubtitle>
        <CTAButtonTitleWrapper>
          <StyledClosedIconButton
            data-testid='close-dialog-button'
            onClick={(e) => {
              goToChooseDestionationView()
            }}
          />
          <StyledTitleHeading headingType={'h2'} fontSize={28}>
            Your Trip
          </StyledTitleHeading>
        </CTAButtonTitleWrapper>
        <Paragraph fontSize={14}>
          To kepp track of your trips, a good idea is to name the specific trip
          you are about to create. It's optional but a good idea.
        </Paragraph>
        <StyledParagraph fontSize={18}>Trip name</StyledParagraph>
        <Input />
      </TopCTAButtonTitleSubtitle>
      <StyledButton onClick={createTrip}>Create Commute trip</StyledButton>
    </React.Fragment>
  )
}

export default NameCommuteView
