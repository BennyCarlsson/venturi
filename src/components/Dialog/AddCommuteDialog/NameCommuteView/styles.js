import styled from 'styled-components'
import Button from 'components/Button'
import Heading from 'components/Heading'
import Paragraph from 'components/Paragraph'
import ClosedIconButton from 'components/CloseIconButton'

export const TopCTAButtonTitleSubtitle = styled.div`
  margin-top: 8px;
`

export const CTAButtonTitleWrapper = styled.div`
  display: flex;
  align-items: baseline;
  margin-bottom: 16px;
`

export const StyledParagraph = styled(Paragraph)`
  padding-top: 48px;
`

export const StyledTitleHeading = styled(Heading)`
  margin-left: 24px;
`
export const Input = styled.input`
  position: relative;
  z-index: 1;
  width: 100%;
  height: 36px;
  border: none;
  border-radius: 0;
  border-bottom: 1px solid #726a95;
  font-family: 'Montserrat';
  font-style: normal;
  font-size: 14px;
  line-height: 17px;
  padding: 8px;
`
export const StyledButton = styled(Button)`
  margin-bottom: 56px;
`

export const StyledClosedIconButton = styled(ClosedIconButton)``
