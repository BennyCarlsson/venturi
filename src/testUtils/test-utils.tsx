import { configureStore } from '@reduxjs/toolkit';
import locationsReducer from '../redux/locationsSlice';
import testReducer from '../redux/testSlice';
import tripReducer from '../redux/tripSlice';
import dialogReducer from '../redux/dialogSlice';
import { render as rtlRender, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ReactElement } from 'react';
import { RootState } from 'redux/store';

type RenderOptionsType = { initialState?: Partial<RootState> } & Omit<RenderOptions, 'queries'>;
function render(ui: ReactElement, { initialState = {}, ...renderOptions }: RenderOptionsType) {
  const store = configureStore({
    reducer: {
      test: testReducer,
      trip: tripReducer,
      locations: locationsReducer,
      dialog: dialogReducer
    },
    preloadedState: initialState
  });
  function Wrapper({ children }: any) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
