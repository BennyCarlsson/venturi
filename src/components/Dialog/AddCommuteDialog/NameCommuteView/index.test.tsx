import userEvent from "@testing-library/user-event";
import { render, screen } from "testUtils/test-utils";
import NameCommuteView from ".";

describe("<NameCommuteView />", () => {
  const mockCreateTrip = jest.fn();
  const mockGoToChooseDestionationView = jest.fn();

  beforeEach(() => {
    render(
      <NameCommuteView
        createTrip={mockCreateTrip}
        goToChooseDestionationView={mockGoToChooseDestionationView}
      />,
      {}
    );
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <NameCommuteView
        createTrip={mockCreateTrip}
        goToChooseDestionationView={mockGoToChooseDestionationView}
      />,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    render(
      <NameCommuteView
        createTrip={mockCreateTrip}
        goToChooseDestionationView={mockGoToChooseDestionationView}
      />,
      {}
    );
    expect(screen).toBeTruthy();
  });

  it("should call mockCreatTrip on", () => {
    expect(mockCreateTrip).not.toHaveBeenCalled();
    userEvent.click(
      screen.getByRole("button", { name: "Create Commute trip" })
    );
    expect(mockCreateTrip).toBeCalledTimes(1);
  });

  it("should call mockGoToChoseDestinationView", () => {
    expect(mockGoToChooseDestionationView).not.toHaveBeenCalled();
    userEvent.click(screen.getByTestId("close-dialog-button"));
    expect(mockGoToChooseDestionationView).toBeCalledTimes(1);
  });
});
