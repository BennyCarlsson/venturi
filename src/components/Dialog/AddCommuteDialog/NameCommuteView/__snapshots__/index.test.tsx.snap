// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NameCommuteView /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jrAGrp jwoKOl"
  >
    <div
      class="sc-kEjbxe kbqPt"
    >
      <button
        class="sc-fubCfw cfizV sc-kstrdz"
        data-testid="close-dialog-button"
      >
        <svg
          class="sc-pFZIQ hzDuJt"
        >
          chevron-down.svg
        </svg>
      </button>
      <h2
        class="sc-hKgILt jpaqbD sc-crrsfI jplDhQ"
        font-size="28"
      >
        Your Trip
      </h2>
    </div>
    <p
      class="sc-bdfBwQ QuMsF"
      font-size="14"
    >
      To kepp track of your trips, a good idea is to name the specific trip you are about to create. It's optional but a good idea.
    </p>
    <p
      class="sc-bdfBwQ kMVXxV sc-iqHYGH JPwUs"
      font-size="18"
    >
      Trip name
    </p>
    <input
      class="sc-dQppl gUpTvy"
    />
  </div>
  <button
    class="sc-gsTCUz jMcfyH sc-bqyKva dotLk"
  >
    Create Commute trip
  </button>
</DocumentFragment>
`;
